### ISSUE TITLE
clean urls using rewrite rules for lighttpd

### ISSUE THREAD

**Issue by abhayghatpande** [justinwalsh/daux.io#380]
I used this https://www.drupal.org/node/139343 for reference, to try to write similar rules for daux.io, but no success. 

Has someone used daux with lighttpd? Any pointers how to get it to work?

**Comment by abhayghatpande** [comment]
OK. So got it to work partially at least thanks to this post: https://www.guyrutenberg.com/2008/05/24/clean-urls-permalinks-for-wordpress-on-lighttpd/

The only way I got this to work was to set the document root to the daux directory in the `/etc/lighttpd/lighttpd.conf`:
`server.document-root        = "/var/www/daux/"`
and then the rest is exactly the same as the post.

```
url.rewrite-once = (
        "^/(.*)\.(.+)$" => "$0",
        "^/(.+)/?$" => "/index.php/$1"
    )
```

The only strange part is that none of the images are showing up.

**Comment by abhayghatpande** [comment]
Figured it out finally. Here is the code to add to `\etc\lighttpd\lighttpd.conf`. Make sure that `mod_rewrite` is active.

```
$HTTP["host"] =~ "<daux server name>" {
server.document-root = "/var/www/daux/"
    url.rewrite-if-not-file = (
        "^/(.*)\.(.+)$" => "/index.php/$0",
        "^/(.+)/?$" => "/index.php/$1"
    )

< other server config like log files, etc.>
}
```

In my case, I unzipped the `daux.io-master.zip` file under `/var/www/daux/`. The server name can be anything that is resolvable by dns like `daux.example.org` as long as the dns settings for the domain are configured correctly. Since Lighttpd does not support `.htaccess` type functionality for individual sites, all the configuration has to be centralized in the `lighttpd.conf` file.

Lighttpd now offers the `url.rewrite-if-not-file` option that is similar to the `-f` and `-d` option of `.htaccess` for Apache.

The first rule checks for files likes images, css, etc., that have a `.` in their names and these are passed unchanged to `index.php`. The 2nd rule passes other files with "clean" names to index.php.

These rules seem to work for directories and subdirectories. I haven't really been able to test it thoroughly, so there may be some edge cases that do not work.

Hope this helps somebody. Maybe it could be added to the current documentation to add support for Lighttpd.
